src/shell/interpreter.zig

```diff
@@ -885,18 +885,11 @@ pub const Interpreter = struct {
         }
 
         pub fn getHomedir(self: *ShellState) EnvStr {
-            if (comptime bun.Environment.isWindows) {
-                if (self.export_env.get(EnvStr.initSlice("USERPROFILE"))) |env| {
-                    env.ref();
-                    return env;
-                }
-            } else {
-                if (self.export_env.get(EnvStr.initSlice("HOME"))) |env| {
-                    env.ref();
-                    return env;
-                }
-            }
-            return EnvStr.initSlice("unknown");
+            const env_var: ?EnvStr = brk: {
+                const static_str = if (comptime bun.Environment.isWindows) EnvStr.initSlice("USERPROFILE") else EnvStr.initSlice("HOME");
+                break :brk self.shell_env.get(static_str) orelse self.export_env.get(static_str);
+            };
+            return env_var orelse EnvStr.initSlice("unknown");
```

Comment:
does it make sense to return the string `unknown`? If we expand `~` to `unknown` it's just going to lead to really confusing errors